### DISCUSSION TITLE
collapsing (hiding) certain lines on text (not code) document?

### DISCUSSION THREAD

**Discussion by dr-dba** [orbitalquark/textadept#59] [General]
Hello, i did some post into the ScintiLua topic before, 
but probaly here is more appropriate place to ask the question and advice.
I need ad-hoc collapsing/hiding of text lines (not code folding).
I see it can be done by putting special symbols and parse that with lexer,
as here suggested in a classic SciTE article:
http://lua-users.org/wiki/SciteTextFolding
.. but i dont like the idea typing/modifying the text only for folding

How is it possible in TextAdept to achieve proper seamless lines collapsing?

**Comment by orbitalquark** [comment]
Yes, you can use `view:hide_lines()`[1] and `view:show_lines()`[2]. You can input these commands directly in the command entry, or you could call them from a function that is activated via key binding, mouse click, etc.

[1]: https://orbitalquark.github.io/textadept/api.html#view.hide_lines
[2]: https://orbitalquark.github.io/textadept/api.html#view.show_lines

**Reply by dr-dba** [reply]
Great, sorry for my blindness. looks exactly what is needed.
i dont know how i missed that, looking a few times, and searching for "hide", and so on.

**Comment by orbitalquark** [comment]
No problem. It's hard to find unless you know what you're looking for.

**Comment by dr-dba** [comment]
Just in case, i will mention i want to make some GREPping for the open text document, 
that is to say, temporary hide lines not relevant, leave lines relevant,
surrounded with line neighbores above and below, kind of framing.
And probably it done by somebody already? 
That is good also to know and adopt as well

Should i open new discussion/question, or leave that at that?
